backend/workflow_manager/workflow_v2/file_history_helper.py

```
@@ -255,35 +256,90 @@ def create_file_history(
+                logger.info(
+                    f"Retrieved existing FileHistory record after constraint violation - "
+                    f"ID: {existing_record.id}, workflow={workflow}"
+                )
+                return existing_record
+            else:
+                # This should rarely happen, but if we can't find the existing record,
+                # log the issue and re-raise the original exception
+                logger.error(
+                    f"Failed to retrieve existing FileHistory record after constraint violation - "
+                    f"file_name='{file_hash.file_name}', workflow={workflow}"
+                )
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)